src/test/scala/nl/biopet/utils/CountsTest.scala

```diff
@@ -189,4 +190,80 @@ class CountsTest extends BiopetTest {
     c1.acumolateCounts() shouldBe Map(1 -> 1, 2 -> 3, 3 -> 6)
     c1.acumolateCounts(true) shouldBe Map(1 -> 6, 2 -> 5, 3 -> 3)
   }
+
+  @Test
+  def testToDoubleArray(): Unit = {
+    val c1 = new Counts[String](Map("1" -> 1, "2" -> 2, "3" -> 3))
+    val doubleArray = c1.toDoubleArray
+    doubleArray.values.toSet shouldBe Set("1", "2", "3")
```

Comment:
That is true. But in a map the order is not kept. (see c1)
So it is impossible to know beforehand what the order is going to be .
The `Counts.fromDoubleArray` method is used to check if it can be converted back properly.

Also below there are tests that explicitly test order.